### PR TITLE
[FEAT] Megastore UI Update.

### PR DESCRIPTION
**Pull request by LittleEins**
# Description

**New UI**

Introducing a tiered system where players must craft 3/4 items before they can unlock more advanced items.
image

Fixes #issue

# What needs to be tested by the reviewer?

Connect BE. Buy 2 basic items. See if Rare Item Unlocks. Buy Key and see if it locks due to daily limit. 

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/world/scenes/PlazaScene.ts [hunk 1]
```diff
@@ -123,6 +123,11 @@ export const PLAZA_BUMPKINS: NPCBumpkin[] = [
     npc: "chase",
     direction: "left",
   },
+  {
```
Comment (adamhannigan) [inline src/features/world/scenes/PlazaScene.ts:126]: What is the plan with Test Stella - Is it intended to release to the public?

**Review by adamhannigan — CHANGES_REQUESTED** [review]
Nice work @LittleEins 

Just a few small touch ups.

1. I would suggest hiding the count. It feels a bit glitching when moving your mouse around and it flickers on and off. The 'check' is more important than the count.

Screenshot 2024-10-22 at 3 04 17 PM

2. If you navigate to the Seasonal section of the codex, you are able to buy the items in the Megastore. Players should have to visit the Plaza to buy these items. I would suggest removing the "Buy" button if opened up from the Codex.

Screenshot 2024-10-22 at 3 06 09 PM

3. How do we plan on testing and turning this on for players? I tried turning the flag off `SEASONAL_TIERS: () => false` and I was still able to visit the new feature through the 3rd tab. We will want to ensure that the public does not get access to this until the 1st of November.

Screenshot 2024-10-22 at 2 53 41 PM

File: src/features/game/events/landExpansion/buySeasonalItem.test.ts [hunk 2]
```diff
@@ -119,4 +119,56 @@ describe("buySeasonalItem", () => {
...
+          },
+        },
+        action: {
+          type: "seasonalItem.bought",
+          name: "Treasure Key",
+          tier: "basic",
+        },
+        createdAt: new Date("2024-08-09").getTime(),
+      }),
+    ).toThrow("Already bought today");
```
Comment (brian-man) [inline src/features/game/events/landExpansion/buySeasonalItem.test.ts:149]: Maybe the user-facing shopping UX needs to have an indicator (and/or text) that some items have a once-per-day limit?

File: src/lib/npcs.ts [hunk 3]
```diff
@@ -106,7 +106,7 @@ export type NPCName =
   | "pharaoh"
   | "worried pete"
   | "chase" //cowboy
-  | "sheep"; //sheep
+  | "sheep"; //sheep;
```
Comment (brian-man) [inline src/lib/npcs.ts:109]: Lint issue?

**Comment by brian-man** [comment]
Great to see you here, @LittleEins!

File: src/features/game/events/landExpansion/buySeasonalItem.test.ts (hunk 2 above)
Comment (adamhannigan) [reply src/features/game/events/landExpansion/buySeasonalItem.test.ts:149]: I think this related to the keys logic - there is currently a "Daily limit - 1/1" label that helps out 😄

**Pull request merged**